File: Presentation/SmartWeather.WebClient/src/services/station/dtos/station_response.ts
import { ComponentResponse } from "@/services/component/dtos/component_response"

export interface StationDto {
    id: number
    name: string
    latitude: number
    longitude: number
    type: StationType
    userId: number
    components: ComponentResponse[] | undefined
}

export enum StationType {
    Professionnal,
    Private
}


export interface StationLatencyDetailsDto {
    station: StationDto;
    latency: number | undefined;
  }

File: Application/SmartWeather.Api/Controllers/Station/Dtos/StationResponse.cs
using SmartWeather.Api.Controllers.Component.Dtos;
using SmartWeather.Entities.Station;

namespace SmartWeather.Api.Controllers.Station.Dtos;

public class StationResponse
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public float Latitude { get; set; }
    public float Longitude { get; set; }
    public StationType Type { get; set; }
    public int? UserId { get; set; }
    public List<ComponentResponse>? Components { get; set; }
}
